src/NosCore.GameObject/Providers/InventoryService/InventoryService.cs

```diff
@@ -474,6 +474,33 @@ public InventoryItemInstance RemoveItemAmountFromInventory(short amount, Guid id
             return null;
         }
 
+        public void RemoveItemAmountFromInventoryByVNum(short amount, short itemVNum)
+        {
+            var itemInstancesOfVNum = this.Select(s => s.Value).Where(w => w.ItemInstance.ItemVNum == itemVNum)
+                .OrderBy(o => o.ItemInstance.Amount).ToList();
+            var stayAmount = amount;
+            foreach (var itemInstance in itemInstancesOfVNum)
+            {
+                if (stayAmount == 0)
+                {
+                    return;
```

Comment:
Don’t add stuff in here you can do this by searching for the list to delete and then delete them.